models/experimental/functional_stable_diffusion/tt2/ttnn_functional_transformer_2d_new_conv.py

```diff
@@ -314,13 +358,51 @@ def __call__(
         out_channels = in_channels if out_channels is None else out_channels
         if is_input_continuous:
             if not use_linear_projection:
-                if hidden_states.memory_config() != self.proj_out.conv.input_sharded_memory_config:
-                    assert False
+                # if hidden_states.memory_config() != self.proj_out.conv.input_sharded_memory_config:
+                #     assert False
                 # hidden_states = ttnn.to_memory_config(hidden_states, self.proj_out.conv.input_sharded_memory_config)
-                hidden_states = self.proj_out(hidden_states)
-                if ttnn.get_memory_config(residual) != self.proj_out.conv.input_sharded_memory_config:
-                    residual = ttnn.to_memory_config(residual, self.proj_out.conv.input_sharded_memory_config)
-
+                # hidden_states = self.proj_out(hidden_states)
+                [hidden_states, _out_height, _out_width, _dev_weights, _dev_bias] = ttnn.conv2d(
+                    input_tensor=hidden_states,
+                    in_channels=self.proj_out_in_channels,
+                    out_channels=self.proj_out_out_channels,
+                    kernel_size=(1, 1),
+                    stride=(1, 1),
+                    padding=(0, 0),
+                    device=self.device,
+                    batch_size=self.batch_size,
+                    input_height=self.input_height,
+                    input_width=self.input_width,
+                    weight_tensor=self.proj_out_conv_weights,
+                    bias_tensor=self.proj_out_conv_bias,
+                    conv_config=conv_config,
+                    conv_op_cache=conv_cache,
+                )
+                # if ttnn.get_memory_config(residual) != self.proj_out.conv.input_sharded_memory_config:
+                #     residual = ttnn.to_memory_config(residual, self.proj_out.conv.input_sharded_memory_config)
+                print(
+                    "Transformer Add Input 1",
+                    hidden_states.shape,
+                    hidden_states.layout,
+                    hidden_states.memory_config(),
+                    "sharded = ",
+                    hidden_states.is_sharded(),
+                )
+                print(
+                    "Transformer Add Input 2",
+                    residual.shape,
+                    residual.layout,
+                    residual.memory_config(),
+                    "sharded = ",
+                    residual.is_sharded(),
+                )
+                hidden_states = ttnn.experimental.tensor.sharded_to_interleaved(
```

Comment:
Why are we setting both to interleaved?

Reply:
I was facing an error where one was inputs to the addition was block sharded in col major and the other was block sharded in row major, and so they could not be added together. 

The solution I used was to make them both interleaved.

Reply:
Which input was col major? They should both be row major, perf will suffer otherwise. Is the new conv row major for wormhole?

Reply:
I've removed the extra sharded_to_interleaved ops. Instead, I've set the transpose_shards option to False, which outputs row major.